### PR TITLE
Better types

### LINKED ISSUE #3: better types
I used "any" in many places, better fix it

### PR DESCRIPTION
**Pull request by sahariko**
Attempt creating more comprehensive types.

Resolves #3.

### CODE REVIEW

File: src/spy-performance/spy-functions/spy-selector.ts [hunk 1]
```diff
@@ -31,25 +33,25 @@ const spySelectorMemoizer = (originalMemoized: any) => {
   };
 };
 
-export const spyCreateSelectorTime = function (originalCreateSelectorFunc: any) {
-  return function () {
-    arguments[0] = spySelectorMemoizer(arguments[0]);
-    const spiedFunction = originalCreateSelectorFunc(...arguments);
+export const spyCreateSelectorTime = function (originalCreateSelectorFunc: GenericFunction) {
+  return (...args: any[]) => {
```
Comment (sahariko) [inline src/spy-performance/spy-functions/spy-selector.ts:37]: TS does NOT play nice with `arguments`.

The alternative would be to add `@ts-ignore` in certain places.

**Comment by mentaman** [comment]
awesomeee! merging!

had some conflicts so I'll merge to a separate branch first to fix them

File: src/spy-performance/summary/print-summary.ts [hunk 2]
```diff
@@ -16,21 +16,25 @@ export const perfStatsReset = function () {
 };
 
 export const getPerfSummary = function () {
-  const longestSelectors = Object.entries(selectorsPerfStat.stats)
-    .map(([key, value]) => ({ ...value, key }))
-    .sort((a, b) => b.duration - a.duration);
+  const a = Object.entries(selectorsPerfStat.stats)
```
Comment (mentaman) [inline src/spy-performance/summary/print-summary.ts:19]: what are those 4 variables?
removed them for now, let me know if they're needed

Comment (sahariko) [reply src/spy-performance/summary/print-summary.ts:19]: My bad, was an attempt to figure out what was going wrong with types 🙈

**Pull request merged**